core/pkg/monitor/gpu_nvidia.go

```diff
@@ -48,19 +48,23 @@ func isRunning(cmd *exec.Cmd) bool {
 	return err == nil
 }
 
+// GPUNvidia monitors NVIDIA GPU stats using the nvidia_gpu_stats command.
 type GPUNvidia struct {
 	name             string
 	sample           map[string]any // latest reading from nvidia_gpu_stats command
 	pid              int32          // pid of the process to monitor
 	samplingInterval float64        // sampling interval in seconds
 	lastTimestamp    float64        // last reported timestamp
 	mutex            sync.RWMutex
+	cmdPath          string
 	cmd              *exec.Cmd
 	logger           *observability.CoreLogger
 	waitOnce         sync.Once
 }
 
-func NewGPUNvidia(logger *observability.CoreLogger, pid int32, samplingInterval float64) *GPUNvidia {
+// NewGPUNvidia creates a new GPUNvidia instance configured to monitor NVIDIA GPUs.
+// If cmdPath is empty, it will use the default nvidia_gpu_stats path.
+func NewGPUNvidia(logger *observability.CoreLogger, pid int32, samplingInterval float64, cmdPath string) *GPUNvidia {
```

Comment:
nit:

* the function signature is too long; break it into one parameter per line with a trailing comma
* docs should have a blank line (`//`) between the summary sentence and any details